### ISSUE TITLE
Bob is not working with postgresql array

### ISSUE THREAD

**Issue by hiendaovinh** [stephenafamo/bob#90]
Hi, I was trying to use bob with pgx. When I try to fetch items from QueryContext it gave me this error
```
pq: unable to parse array; expected '{' at offset 0
```

I believe this is a problem with scanner. Please take a look.

**Comment by stephenafamo** [comment]
My immediate guess is that this is happening because the array types in pgx do not implement `sql.Scanner`.  

If using `github.com/stephenafamo/scan` directly, it is possible to add the custom pgx types using `WithScannableTypes`. However, the code generated by Bob does not account for this.

**Comment by hiendaovinh** [comment]
I believe this will be fixed soon.

Ref:
https://github.com/golang/go/issues/67546
https://github.com/jackc/pgx/pull/2029
https://github.com/jackc/pgx/issues/1779

**Comment by stephenafamo** [comment]
That would make a lot of things simpler 👍🏾

**Comment by smitpatelx** [comment]
I think it's still not working for inserting arrays in Postgres.
- I'm new to bob orm, so I don't know if their is any inbuilt way to deal with array types. I also didn't found the documentation helpful for new devs learning bob.
- For now, I just wrote a small function which takes array of string and return the pg compatible array.
```go
func getArrayString(stringArray []string) string {
    if len(stringArray) == 0 {
        return "{}"
    }

    result := "{" + stringArray[0]
    for i := 1; i < len(stringArray); i++ {
        result += "," + stringArray[i]
    }
    result += "}"
    return result
}
```

**Comment by stephenafamo** [comment]
@smitpatelx A slice to strings does not automatically map to a Postgres array. You can use a prebuilt type such as [pg.StringArray](https://pkg.go.dev/github.com/lib/pq#StringArray) which will do the conversion when reading from or writing to the database.

**Comment by hiendaovinh** [comment]
@stephenafamo For now, I have to use this as a workaround while working with bob and pgx
```go
q := bob.Categories.Update(mods, um.Where(bob.CategoryColumns.ID.EQ(psql.Arg(id))))

q.Expression.AppendMapperMod(func(ctx context.Context, c []string) (scan.BeforeFunc, scan.AfterMod) {
	return func(v *scan.Row) (any, error) {
			v.ScheduleScan("tags", &pgtype.FlatArray[string]{})
			return nil, nil
		}, func(link, retrieved any) error {
			return nil
		}
})

item, err := q.One(ctx, ds.bobExecutor)
```

Is there anything we could do to customize bobgen with this behavior (mapper for configuration, etc.)?

Couldn't get it working with `types` and `replacements` configuration.
Thanks.

**Comment by stephenafamo** [comment]
In your types configuration, why not replace it directly with `pgtype.FlatArray[string]`? @hiendaovinh

**Comment by hiendaovinh** [comment]
For me, there is no benefit using `pgtype.FlatArray[string]` over `[]string`, therefore, I stick with the more-primitive choice.

**Comment by stephenafamo** [comment]
BTW, it doesn't look to me like your earlier snippet will work.

While the scan is scheduled into the `pgtype.FlatArray[string]` the value is not actually used anywhere. I assume the `[]string` on the struct will just be nil.

**Comment by hiendaovinh** [comment]
Yeah, the snippet won't work. So I have the Update 2 to make everything work without AppendMapperMod.

**Comment by stephenafamo** [comment]
> This is necessary because there is a problem with `null.Val[[]string]` scan. `[]string` works fine.

Can you create a new issue with more details about this?

**Comment by stephenafamo** [comment]
Likely 👍🏾
